runtime/src/hardened_unpack.rs

```diff
@@ -132,6 +132,11 @@ where
         }
 
         let parts: Vec<_> = parts.map(|p| p.unwrap()).collect();
+        let account_filename: Option<PathBuf> = if parts.len() == 2 && parts[0] == "accounts" {
+            Some(PathBuf::from(parts[1]))
+        } else {
+            None
+        };
```

Comment:
A bit more succinct, imo:

```suggestion
        let account_filename = (if parts.len() == 2 && parts[0] == "accounts").map(|| PathBuf::from(parts[1]));
```

(Did not check if this compiles, and needs formatting)

Reply:
Thanks!  Great to learn these options!  I think i will use then_some here because the parameter is simpler.

Reply:
Got it.  Changed to then().